core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -30,15 +32,57 @@ type AssetLockValue struct {
 	ExpiryTimeSecs	uint64	`json:"expiryTimeSecs"`
 }
 
-// function to generate contract id as "SHA256" hash based on the asset lock key (which is combination of asset type and asset it)
-func generateAssetLockKeyAndContractId(assetAgreement *common.AssetExchangeAgreement) (string, string) {
+// Object used in the map, contractId --> <asset-type, num-units, locker, ...> (for fungible assets)
+type FungibleAssetLockValue struct {
```

Comment:
Don't you need a `ContractId` field here just as in `AssetLockValue`? If your data map separates the contract Id from the lock information and uses the former purely as a primary key to look up the contract details, then I would recommend you remove it from `AssetLockValue` too, and use suitable RW sets that enable an O(1) lookup based on this field for either category of asset.

For fungible assets, you can index your storage using <contract id>--><contract lock info>
For non-fungible assets, you can index your storage using <contract id>--><asset type, asset id>--><contract lock info>.
This produces some consistency between the two categories and also allows fast lookup based on <contract id> as well as <asset type, asset id> in the case of non-fungible assets.

Reply:
@VRamakrishna , I have removed `ContractId` field from `AssetLockValue` too as suggested. Yes, the indexing is now as per what you have summarized above and consistent.